tests/test_validators.py

```diff
@@ -22,6 +23,19 @@
 from pydantic.decorators import field_validator, root_validator
 
 
+def test_annotated_validator() -> None:
+    MyInt = Annotated[int, AfterValidator(lambda x: x if x != -1 else 0)]
+
+    def any_key_but_exclude(d: Dict[str, Any]) -> Dict[str, Any]:
+        return {k: v for k, v in d.items() if k != 'exclude'}
```

Comment:
I thnk this could also be done in the item validator using [`PydanticOmit`](https://github.com/pydantic/pydantic-core/blob/f44a94298b4be4241c23f61bd528004c8a7e9777/pydantic_core/_pydantic_core.pyi#L208) without needing to iterate over the dict again.

We should probably check that behaviour rather than this route.

Reply:
This was somewhat of an arbitrary example, I'll change it to something else

Reply:
replaced